CMakeLists.txt

```diff
@@ -105,7 +105,7 @@ set(Boost_USE_MULTITHREADED      ON)
 set( Boost_USE_STATIC_LIBS ON CACHE STRING "ON or OFF" )
 # Most boost deps get implictly picked up via fc, as just about everything links to fc. In addition we pick up
 # the pthread dependency through fc.
-find_package(Boost 1.70 REQUIRED COMPONENTS program_options unit_test_framework system)
+find_package(Boost 1.71 REQUIRED COMPONENTS program_options unit_test_framework system)
```

Comment:
As far as I can tell, Boost 1.71 does not work with C++20 at least for clang++16. Does this need to be 1.74 at least?

Reply:
indeed, when we enable c++20 we're going to need to do.. something else.

Reply:
Yes, we'll have to change the boost minimum version when we enable c++20.
I personally would like us to use boost 1.82 always. Maybe add `boost` as a submodule (I talked with Areg about it already).
I think we would benefit to use a common version of boost regardless of how (pinned vs unpinned) or on what system `leap` is built on. 

This would also allow us to use newer libraries such as `boost::json`.